### PR TITLE
prevent crash after loading : 1st nrrd image, 2: nhdr, 3 : nrrd, 

### PR DESCRIPTION
**Pull request by nbaraka**
I do not knwo why there this issue. 
Because if  you load first nhdr image then you do not this pb. 
I may hide a bug by preventing this crash. (smthing related LayerInfoVec in vtkImageView3D)

### CODE REVIEW

File: src/layers/legacy/medVtkInria/vtkImageView/vtkImageView3D.cxx [hunk 1]
```diff
@@ -531,7 +531,8 @@ void vtkImageView3D::InternalUpdate()
         // append all scalar buffer into the same image
         for (auto it : this->LayerInfoVec)
         {
-            appender->AddInputConnection(it.ImageDisplay->GetInputProducer()->GetOutputPort());
+            if(it.ImageDisplay->GetInputProducer()!=nullptr)
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkImageView/vtkImageView3D.cxx:534]: can you add {}?

**Comment by mathildemerle** [comment]
Can you explain how to create the bug? What data, order, etc :)

**Comment by nbaraka** [comment]
_> Can you explain how to create the bug? What data, order, etc :)_

I used data from a tutorial "White_Matter_Exploration_Tutorial" : https://bit.ly/3a95Eb9

I load the nrrd,then the nhdr and then an other nrrd file.

**Comment by mathildemerle** [comment]
Weirdly, i can't reproduce the bug. I tried with the 2 files from the White_Matter project:
 * open BaselineVolume nrrd
 * open DTIVolume nhdr
 * in [medInria-data](https://github.com/medInria/medInria-data) (test data for medInria), open ODF/odf.nrrd

I tried on medInria3.1.x, and master. 
Is something missing in my test?

**Comment by nbaraka** [comment]
_> Weirdly, i can't reproduce the bug. I tried with the 2 files from the White_Matter project:
> 
> * open BaselineVolume nrrd
> * open DTIVolume nhdr_
> * in [medInria-data](https://github.com/medInria/medInria-data) (test data for medInria), open ODF/odf.nrrd
> 
> I tried on medInria3.1.x, and master.
> Is something missing in my test?

as third data I used the nrrd from https://bit.ly/2RGHyy5 (since a tutorial), all the data are temporarly imported and opened in the visualization.

**Comment by mathildemerle** [comment]
Indeed, i have the crash with these 3 specific files

**Comment by mathildemerle** [comment]
The Pull Request solves the crash but does not display the third data in 3D. It could be interesting to solve that also

**Comment by nbaraka** [comment]
_> The Pull Request solves the crash but does not display the third data in 3D. It could be interesting to solve that also_
The 3d view issue is related to #606

**Comment by mathildemerle** [comment]
Solved in https://github.com/medInria/medInria-public/pull/623, thx @nbaraka

**Pull request closed without merging**